src/java/org/apache/cassandra/index/sai/disk/vector/CompressedVectorRowIdIterator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.vector;
+
+import java.util.NoSuchElementException;
+import java.util.PriorityQueue;
+import java.util.function.IntFunction;
+
+import io.github.jbellis.jvector.vector.VectorSimilarityFunction;
+import org.apache.cassandra.utils.CloseableIterator;
+
+/**
+ * An iterator over {@link ScoredRowId} that lazily orders a {@link PriorityQueue} of approximately scored rows.
+ * The class works by:
+ * 1. Taking a priority queue of {@link ApproximateScore} ordered by approximate similarity score.
+ * 2. Polling the topK results from the approximate score queue
+ * 3. Materialize and score the full resolution vector for each row from step 2.
+ * 4. Add to the full resolution row {@link ScoredRowId} to the priority queue.
+ * 5. Return rows from the {@link ScoredRowId} priority queue until the limit is reached.
+ * 6. When the limit is reached, if the {@link ApproximateScore} queue is not empty, get the next (topK - limit) rows
+ *    from the approximate score queue and repeat 3-6 until the {@link ApproximateScore} queue is empty.
+ * 7. Return the remaining rows from the {@link ScoredRowId} queue.
+ */
+public class CompressedVectorRowIdIterator implements CloseableIterator<ScoredRowId>
```

Comment:
The current name is fine but I'd lean towards something like `BruteForceRowIdIterator` so that you get a sense of why and not just what